internal/server/routes.go

```diff
@@ -112,6 +112,7 @@ func (s *Server) GenerateRoutes(promRegistry prometheus.Registerer) Routes {
 	get(a, noAuthn, "/api/providers/:id", a.GetProvider)
 
 	get(a, noAuthn, "/api/version", a.Version)
+	get(a, noAuthn, "/api/password", a.GetPasswordSettings)
```

Comment:
This seems like a no-so-good path for password requirements. `password` is not really a resource. In this case, it seems a better path could be `/api/settings` which can be the start of the settings API

Reply:
I'd be fine with /api/password-requirements or something, but I think you have a point about their being a lot of different settings clients will need on startup. it doesn't make a ton of sense to make them query them all separately

Reply:
@mxyng @ssoroka `/api/settings?name=passwordrequirements`?

Reply:
I think we need to consider how this endpoint will be authorized. There are already things in settings that should only be visible to admins (`PrivateJWK`), and I expect there to be more.

Password requirements on the other hand are not sensitive, and will likely need to be queried without authorization.

I think this means that either:
* we'll need the endpoint to filter out some values based on authorization (something I think is fine, but I know @ssoroka has expressed not wanting to do)
* we should have separate endpoints for public and private settings

I don't think we want an endpoint that returns different structures, so I don't think a `name=...` query param is the right solution. If we change the output it should be based on authorization, not query parameters.

Maybe `/api/settings/public` ?

Reply:
We don't currently have a settings API `/api/settings` though you make a great point. There will be public and private settings as well as user and organization settings. These will all need to be kept separate since they represent distinct audiences.